PowerShell Contributing.md

```diff
@@ -0,0 +1,479 @@
+
+(If you want to know which tools are inside, use `Get-Command -Module build`)
+
+The very fist time, you need to set up a few things.
+To do that, run `Start-PsBootstrap` (may need sudo/adminstrative privileges, depending on your system.
+On a docker container, it needs the `-NoSudo` switch).
+
+You also need to build the pwsh binaries for the tests to run against: `Start-PsBuild`
+
+> Tip: If you encounter weird build errors, just run `Start-PSBuild -Clean`.
+This is generally a good idea every time you update your branch, or after a rebase.
+
+Now you could simply run `Start-PSPester` to kick off a build, and run ALL Pester tests.
+This would take some time, and this is still not what you want to do when writing extra tests, or improving existing ones.
+So what do we do instead? We tell `Start-PSPester` which testfile we want to run:
+
+`Start-PSPester -Path .\test\powershell\Modules\Microsoft.PowerShell.Core\Get-Command.Tests.ps1`
+
+You will notice that it rebuilds test tool binaries each time you use Start-PSPester.
+This takes time and usually we only need to build these binaries once.
+To suppress new builds, you add the `-SkipTestToolBuild` switch:
+
+`Start-PSPester -Path .\test\powershell\Modules\Microsoft.PowerShell.Core\Get-Command.Tests.ps1 -SkipTestToolBuild`
+
+When writing and debugging your tests, remember that you can set breakpoints and step through your Pester code, this can help you find out where things are not going as they | Should -Be :-).
+
+> Tip: If you want to learn more about VSCode debugging for PowerShell, you can read more [here on the Microsoft scripting blog](https://devblogs.microsoft.com/scripting/debugging-powershell-script-in-visual-studio-code-part-1/), or watch [this video](https://youtu.be/cSbIXmlkr8o) from Tyler Leonhardt at PSconfEU 2019.
+
+For specific tips about how to use Pester within the PowerShell repo, be sure to read this document as an addendum to the Pester documentation: <https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md>
+
+### Updating your fork
+
+So you have forked one of the GitHub repos, cloned it, tried stuff, but never created a PR. And now that fork is six months behind its ancester. How to update it? I'll describe how to do this for the PowerShell repo, this works the same for the other repos.
+
+To update your GitHub fork, you will first update your local clone, and for that, you need to tell your local repository to which remote to look. 
+
+- See which remotes are configured for your local repo: `git remote -v`
+- If there is an entry with `https://github.com/PowerShell/PowerShell.git` in it, remember its name. It will usually be called 'upstream'
+- If there is no entry present with that url, add it by typing:
+`git remote add upstream https://github.com/PowerShell/PowerShell.git`
+- Make sure there is an entry 'origin' with the url of your own GitHub url: `https://github.com/PowerShell/PowerShell.git` 
```

Comment:
```suggestion
- Make sure there is an entry 'origin' with the url of your own GitHub url: `https://github.com/<YOUR_GITHUB_NAME>/PowerShell.git` 
```

Reply:
Ah! This. You read what I intended to write :) good catch.